### DISCUSSION TITLE
[BE] 6차 데모데이 요구사항 업무 분담 등

### DISCUSSION THREAD

**Discussion by zeus6768** [woowacourse-teams/2024-code-zap#726] [이야기해봐요]
# 안건

> 1. 인스턴스 한 대 기준으로 핵심기능에 목표 TPS를 정한다. 그리고 안정적으로 서비스될 수있게 아래 값을 설정하고 이유를 공유해야한다.
> 2. 톰캣 설정 중 아래 값을 상황에 맞게 설정
> - 2\. threads max
>   - max connections
>   - accept count
> - 3\. hikariCP 커넥션풀 설정
>   - 수치를 설정하고 정한 이유를 발표
>   - hikariCP configuration 보고 필요한 값 설정
> 4. 무중단 배포를 구현하고 배포 과정을 발표한다.
> - 서비스를 배포하는 중간에도 사용자는 서비스를 계속해서 사용할 수 있어야 한다.
> 5. 현재 팀의 아키텍처에서 사용자가 늘어났을 때 문제가 될 수 있는 부분을 설명하고, 이를 개선할 수 있는 방법에 대해 작성한다.
> - 만명, 10만명, 1,000만명이 사용해도 서비스할 수 있도록 팀의 인프라 아키텍처 개선 방향을 공유한다.

# 결정

10/08 (화) - 무중단 배포 어떻게 할지 고민해오기 ✅ - [관련 디스커션](https://github.com/woowacourse-teams/2024-code-zap/discussions/746)
10/11 (금) - 톰켓 설정 고민해오기

**Comment by jminkkk** [comment]
클선생님한테 쪼개달라함

# 1주차: 성능 분석 및 최적화 준비

## 목표 TPS 설정 및 현재 시스템 분석 (2-3일)

서비스의 핵심 기능 파악 및 우선순위 지정
적절한 목표 TPS 설정
현재 시스템의 성능 측정 및 기준점 설정


## 톰캣 및 HikariCP 설정 연구 (2-3일)

threads max, max connections, accept count에 대한 깊이 있는 연구
HikariCP 설정 옵션 및 최적화 전략 조사
다양한 설정 시나리오 작성


## 초기 성능 테스트 환경 구축 (2일)

성능 테스트 도구 선택 및 설정 (예: JMeter, Gatling)
테스트 시나리오 작성



# 2주차: 설정 최적화 및 무중단 배포 구현

## 톰캣 및 HikariCP 설정 최적화 (3일)

1주차 연구를 바탕으로 최적 설정값 적용
각 설정값에 대한 상세한 근거 문서화


## 성능 테스트 및 반복적 조정 (2일)

최적화된 설정으로 성능 테스트 수행
결과 분석 및 추가 조정


## 무중단 배포 전략 수립 및 구현 (4일)

무중단 배포 전략 선택 및 상세 설계
배포 스크립트 작성 및 테스트 환경 구축
초기 구현 및 테스트



# 3주차: 확장성 분석, 개선 및 마무리

## 무중단 배포 테스트 및 최적화 (2일)

실제 환경과 유사한 조건에서 무중단 배포 테스트
문제점 식별 및 개선


## 확장성 분석 및 개선 방안 수립 (3일)

현재 아키텍처의 상세한 병목 지점 분석
사용자 증가 시나리오별 (만명, 10만명, 1,000만명) 상세 확장 전략 수립
개선된 인프라 아키텍처 설계 및 검증


## 최종 문서화 및 발표 준비 (3일)

모든 작업 결과의 종합적 정리 및 문서화
발표 자료 작성 및 리허설
팀 내 최종 리뷰 및 피드백 반영


## 예비 시간 및 추가 최적화 (2일)

예상치 못한 문제 해결을 위한 버퍼 시간
필요시 추가적인 성능 최적화 또는 아키텍처 개선
